### ISSUE TITLE
Run the pnpm service in the background

### ISSUE THREAD

**Issue by zkochan** [pnpm/pnpm#960]
Once #832 is done, we'd like an option to run the service in the background, automatically.

Possibly the https://github.com/IonicaBizau/diable can be used to achieve this

**Comment by zkochan** [comment]
So I tried the `diable` by Ionica Bizau and it works like a charm!

The only issue is that once the server runs in the background, there is no easy way to stop it.

There will be a new command: `pnpm server stop` for stopping the store server.

`pnpm server` will be renamed to `pnpm server start`.

`pnpm server` will print help for the server commands.

**Comment by etamponi** [comment]
Will there be a way to still run the server on the foreground?

Il 15 gen 2018 01:20, "Zoltan Kochan" <notifications@github.com> ha scritto:

**Comment by zkochan** [comment]
I thin `pnpm server start` will still run it in the foreground by default. There will be a flag to run a background server on `pnpm install`. And another flag for `pnpm server`, something like `pnpm server start --background`

**Comment by etamponi** [comment]
Cool. Another important thing: if the server is running in the foreground, `pnpm server stop` should not do anything, or at least I need a flag to disable it. Otherwise, in Glitch, any user can stop the pnpm server globally for everyone 😨

**Comment by zkochan** [comment]
Interesting, maybe we can have something like `pnpm server start --unstoppable`

**Comment by vjpr** [comment]
Running `pnpm server` while an existing server is running shows this:

```
pnpm server
 WARN  The store server is an experimental feature. Breaking changes may happen in non-major versions.
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE /Users/Vaughan/.pnpm-store/2/socket
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at Server.setupListenHandle [as _listen2] (net.js:1334:19)
    at listenInCluster (net.js:1392:12)
    at Server.listen (net.js:1487:5)
    at Object.default_1 [as createServer] (/Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/@pnpm/server/0.6.2/@pnpm!logger@1.0.0/node_modules/@pnpm/server/lib/createServer.js:109:27)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/1.29.1/node_modules/pnpm/lib/cmd/server.js:42:29)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/1.29.1/node_modules/pnpm/lib/cmd/server.js:4:58)
    at <anonymous>
```

A more helpful error would be better.

**Comment by vjpr** [comment]
Also, is there anyway to see that the server is being used? Some kind of log messages?

My expectation would be that running `pnpm server status` would bring stdout to foreground, and it would show me a message when the server is running an install.

**Comment by zkochan** [comment]
I think `pnpm server status` can print some info about the server. Like whether it runs or not, what port it uses, etc.

`pnpm server attach` can bring the server to the foreground (similarly as [docker attach](https://docs.docker.com/engine/reference/commandline/attach/#description))

**Comment by zkochan** [comment]
:ship: published with `pnpm@1.30.0`

**Comment by zkochan** [comment]
Created follow-up issues: #997, #998
